src/internal/operations/backup.go

```diff
@@ -154,11 +163,139 @@ func (op *BackupOperation) Run(ctx context.Context) (err error) {
 	return err
 }
 
+// calls kopia to retrieve prior backup manifests, metadata collections to supply backup heuristics.
+func produceManifestsAndMetadata(
+	ctx context.Context,
+	kw *kopia.Wrapper,
+	sw *store.Wrapper,
+	sel selectors.Selector,
+	acct account.Account,
+) ([]*snapshot.Manifest, []data.Collection, error) {
+	complete, closer := observe.MessageWithCompletion("Fetching backup heuristics:")
+	defer func() {
+		complete <- struct{}{}
+		close(complete)
+		closer()
+	}()
+
+	m365, err := acct.M365Config()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var (
+		tid         = m365.AzureTenantID
+		oc          = selectorToOwnersCats(sel)
+		ms          = kopia.FetchPrevSnapshotManifests(ctx, kw.Conn(), &oc)
+		collections []data.Collection
+	)
+
+	for _, man := range ms {
+		bup := backup.Backup{}
+		if err := sw.Get(
+			ctx,
+			model.BackupSchema,
+			model.StableID(man.Tags[kopia.BackupIDTag]),
+			&bup,
+		); err != nil {
+			return nil, nil, err
+		}
+
+		for _, fn := range []string{graph.DeltaTokenFileName} {
+			for ro := range oc.ResourceOwners {
+				for _, sc := range oc.ServiceCats {
+					colls, err := collectMetadata(ctx, kw, fn, tid, ro, bup.SnapshotID, sc)
+					if err != nil {
```

Comment:
not finding a metadata file is not necessarily an error. This could happen in cases where a backup contains only some of the services we're now backing up (in which case we may have multiple base snapshots). For example `backup create exchange --data email` and `backup create exchange --data contacts,events` followed by `backup create exchange`

As long as all of the metadata files we need are present (i.e. both delta tokens and `map[container ID]container path` _for each category_ in the future) then we can run more efficiently. If one or more are missing for a category we'll need to fallback to doing a full backup of the data

After explaining this though, it seems like we'll probably want to augment the return value of `FetchPrevSnapshotManifests` to also tell us _why_ a specific snapshot was picked. That will allow us to properly choose what metadata to extract from the snapshot. In turn, that will help avoid the case where we have two snapshots that contain metadata for the same category but we only want to use the most recent version

Ex.
```
backup create exchange --data email,contacts -> B1
backup create exchange --data email -> B2
// should use B1 to source contacts metadata, B2 to source emails metadata
backup create exchange
```

Reply:
The "why" a snapshot was picked could probably be denoted with tags or something (ex. this resource owner and serviceCat combo), but may require an update to how we fetch snapshots since that code makes aggressive use of caching

Vaguely related, that code could also probably be simplified if we start tagging checkpoints with an additional tag. Then we wouldn't have to lookup so many manifests to determine if they were complete or not (see #1672)